app/models/ownership.rb

```diff
@@ -12,7 +21,54 @@ def self.by_indexed_gem_name
       .order("rubygems.name ASC")
   end
 
+  def valid_confirmation_token?
+    token_expires_at > Time.zone.now
+  end
+
+  def generate_confirmation_token
+    self.token = SecureRandom.hex(20).encode("UTF-8")
+    self.token_expires_at = Time.zone.now + Gemcutter::OWNERSHIP_TOKEN_EXPIRES_AFTER
+  end
+
+  def confirm_ownership!
+    update(confirmed_at: Time.current)
+  end
+
+  def confirmed?
+    confirmed_at.present?
+  end
+
+  def unconfirmed?
+    !confirmed?
+  end
+
+  def notify_owner_removed
+    rubygem.notifiable_owners.each do |notified_user|
```

Comment:
As far as I understand `notifiable_owners` is owners who have gem push notification enabled.